src/Servers/HttpSys/src/RequestProcessing/Request.cs

```diff
@@ -428,6 +430,13 @@ private void GetTlsHandshakeResults()
         HashStrength = (int)handshake.HashStrength;
         KeyExchangeAlgorithm = handshake.KeyExchangeType;
         KeyExchangeStrength = (int)handshake.KeyExchangeStrength;
+
+        var sni = RequestContext.GetClientSni();
+        var clientSentSni = !sni.Flags.HasFlag(HttpApiTypes.HTTP_REQUEST_PROPERTY_SNI_FLAGS.HTTP_REQUEST_PROPERTY_SNI_FLAG_NO_SNI);
```

Comment:
This condition is weird, it will return true in the default case `0`. Why not always set `HostName = sni.HostName`? It should be empty if it wasn't sent.

Reply:
Wanted to separate the case where the client didn't send an SNI (I thought SniHostName being null communicates that better). I could change that though if you prefer having an empty string instead of null for safety.

Reply:
Trying to use null vs String.Empty to distinguish between different conditions is discouraged, it's not self-descriptive. Is there a requirement to know if the client didn't support SNI as compared to sending an empty value? I don't think there's a practical difference from the server side, it requires an SNI value or it doesn't.

Reply:
Makes sense, updated. Thanks

Reply:
I would have objected to this if it meant that HttpSys and Kestrel behaved differently, but it looks like [ServerCertificateSelectionCallback uses the empty string in this case](https://github.com/dotnet/runtime/blob/10222f94e5d89b19959117dd9e5b8576ef63f878/src/libraries/System.Net.Security/src/System/Net/Security/SslAuthenticationOptions.cs#L16) too.